src/components/Cart/CartListItem.jsx

```diff
@@ -37,46 +38,58 @@ const CartListItem = ({ cartItem }) => {
                 break;
         }
 
-        if(qty >= 3 && operation === 'increment') {
-            showToast('We are sorry! Only 3 units allowed in each order!', 'error');
-            return;
-        }
-
-        try {
-            setIsOngoingNetworkCall(true);
-            const productUpdatedInCart = await updateProductInCart(_id, token, operation);
-
-            if(productUpdatedInCart) {
-                showToast('Quantity updated successfully!', 'success');
-                cartDispatch({type: 'ADD_TO_CART', payload: {cartItems: productUpdatedInCart, error: false, loading: false}});
-            }
-            else {
-                showToast('Failed to update quantity. Please try again later.', 'error');
-            }
-            setIsOngoingNetworkCall(false);
-        }
-        catch(error) {
-            console.log('Something went really wrong!');
-            setIsOngoingNetworkCall(false);
-        }        
+        setIsOngoingNetworkCall(true);
+        const isUpdatedProductInCart = await callUpdateProductInCart(_id, operation);
+        setIsOngoingNetworkCall(false);
```

Comment:
My bad, this is correct. As this inside async await it would work as expected.
If line 42 is just making an API call without async await block this will not work as expected.
Sorry for misleading, however try to eliminate `isOngoingNetworkCall` state

Reply:
Thank you.

Yes, I shall use the global state variable to handle this.